mailpoet/lib/Automation/Engine/Builder/CreateWorkflowFromTemplateController.php

```diff
@@ -29,7 +29,11 @@ public function createWorkflow(string $slug): Workflow {
       throw UnexpectedValueException::create()->withMessage('Template not found.');
     }
 
-    $this->storage->createWorkflow($template->getWorkflow());
-    return $template->getWorkflow();
+    $workflowId = $this->storage->createWorkflow($template->getWorkflow());
+    $workflow = $this->storage->getWorkflow($workflowId);
+    if (!$workflow) {
+      throw UnexpectedValueException::create()->withMessage('Workflow not found.');
```

Comment:
I think not having a workflow just after creating it should be an `InvalidStateException` from `MailPoet\Automation\Engine\Exceptions` rather than a 4xx `UnexpectedValueException`.

Reply:
Thank you! Fixed in [e40c459](https://github.com/mailpoet/mailpoet/pull/4362/commits/e40c4590446856712f5ec71d12cbed8050680939)